### PR TITLE
[Impeller] compute UVs in vertex stage.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Split the functionality of texture_fill and tiled_texture_fill vertex stages:

* The former will accept uvs as part of the vertex data and pass it to the fragment stage.
* The latter will compute the uv geometry in the vertex stage from the position and the normalizing transform.

With this functionality, we can delete all of the CPU based uv computation code.  Additionally, there is some clean up to remove support for external textures (which use the tiled texture mode) from the texture contents. The external textures are used with an image shader/draw rect by internal engine code and won't go down the drawImageRect path anymore.

### CODE REVIEW

File: impeller/entity/BUILD.gn [hunk 1]
```diff
@@ -79,7 +80,6 @@ impeller_shaders("modern_entity_shaders") {
     "shaders/gradients/radial_gradient_ssbo_fill.frag",
     "shaders/gradients/sweep_gradient_ssbo_fill.frag",
     "shaders/geometry/points.comp",
-    "shaders/geometry/uv.comp",
```
Comment (jonahwilliams) [inline impeller/entity/BUILD.gn:82]: 1 for 1!

File: impeller/entity/shaders/tiled_texture_fill.frag [hunk 2]
```diff
@@ -21,11 +23,18 @@ in highp vec2 v_texture_coords;
...
 
 void main() {
-  frag_color =
-      IPHalfSampleWithTileMode(texture_sampler,   // sampler
-                               v_texture_coords,  // texture coordinates
-                               float16_t(frag_info.x_tile_mode),  // x tile mode
-                               float16_t(frag_info.y_tile_mode)   // y tile mode
-                               ) *
-      float16_t(frag_info.alpha);
+  if (supports_decal == 1.0) {
```
Comment (jonahwilliams) [inline impeller/entity/shaders/tiled_texture_fill.frag:26]: Allows metal, vulkan, and most gles devices to use this shader at no additional cost over the texture_fill shader.

File: impeller/entity/contents/content_context.h [hunk 3]
```diff
@@ -129,10 +129,8 @@ using TexturePipeline =
 using TextureStrictSrcPipeline =
     RenderPipelineHandle<TextureFillVertexShader,
                          TextureFillStrictSrcFragmentShader>;
-using PositionUVPipeline = RenderPipelineHandle<TextureFillVertexShader,
```
Comment (jonahwilliams) [inline impeller/entity/contents/content_context.h:132]: Unused

**Comment by jonahwilliams** [comment]
come on goldens, please be good lol

**Comment by chinmaygarde** [comment]
📿

**Comment by jonahwilliams** [comment]
And we're good

**Review by chinmaygarde — APPROVED** [review]
This works way nicer! I didn't pay super close attention to the deletions on of the calculation on the CPU but trust they were mechanical.

**Pull request merged**